src/Arcus.WebApi.Security/Authentication/Certificates/CertificateAuthenticationFilter.cs

```diff
@@ -33,71 +38,95 @@ public async Task OnAuthorizationAsync(AuthorizationFilterContext context)
             Guard.For<ArgumentException>(() => context.HttpContext.RequestServices is null, "Invalid action context given without any HTTP request services");
 
             IServiceProvider services = context.HttpContext.RequestServices;
-            ILogger logger = GetLoggerOrDefault(services);
+            ILogger logger = services.GetLoggerOrDefault<CertificateAuthenticationFilter>();
 
-            X509Certificate2 clientCertificate = GetOrLoadClientCertificateFromRequest(context.HttpContext, logger);
-            if (clientCertificate == null)
+            var validator = services.GetService<CertificateAuthenticationValidator>();
+            if (validator is null)
             {
-                
-                logger.LogWarning(
-                    "No client certificate was specified in the HTTP request while this authentication filter "
-                    + "requires a certificate to validate on the configured validation requirements");
-
-                context.Result = new UnauthorizedResult();
+                throw new KeyNotFoundException(
+                    $"No configured {nameof(CertificateAuthenticationValidator)} instance found in the request services container. "
+                    + "Please configure such an instance (ex. in the Startup) of your application");
             }
-            else
+
+            if (TryGetClientCertificateFromRequest(context.HttpContext, logger, out X509Certificate2 clientCertificate))
             {
-                var validator = services.GetService<CertificateAuthenticationValidator>();
-                if (validator == null)
+                bool isCertificateAllowed = await validator.IsCertificateAllowedAsync(clientCertificate, services);
+                if (isCertificateAllowed)
                 {
-                    throw new KeyNotFoundException(
-                        $"No configured {nameof(CertificateAuthenticationValidator)} instance found in the request services container. "
-                        + "Please configure such an instance (ex. in the Startup) of your application");
+                    LogSecurityEvent(logger, LogLevel.Information, "Client certificate in request is considered allowed according to configured validation requirements");
```

Comment:
In 90% of the cases, this is not relevant and adds some noise in the telemetry
```suggestion
                    LogSecurityEvent(logger, LogLevel.Trace, "Client certificate in request is considered allowed according to configured validation requirements");
```